### ISSUE TITLE
Error in cms page renderer when there are no url_rewrites for a cms page

### ISSUE THREAD

**Issue by lukeromanowicz** [SnowdogApps/magento2-menu#76]
CMS Page Node renderer doesn't check if a URL for a particular page exists in `pageUrls` array:
https://github.com/SnowdogApps/magento2-menu/blob/1e538845941bbaeaa83a77c0b10aab493ccedfd8/Block/NodeType/CmsPage.php#L151-L168

Apparently, such an event can happen because `pageUrls` array is hydrated from `url_rewrite` table data When a cms page doesn't have any URL rewrites for that store, then its URL is not included in the `pageUrls` array.

https://github.com/SnowdogApps/magento2-menu/blob/1e538845941bbaeaa83a77c0b10aab493ccedfd8/Model/ResourceModel/NodeType/CmsPage.php#L58-L71

In result we're getting:

```
1 exception(s):
Exception #0 (Exception): Notice: Undefined offset: 3 in /var/www/magento/vendor/snowdog/module-menu/Block/NodeType/CmsPage.php on line 158
```

A solution would be to back fetched data up with `identifier` from `cms_page` joined with `cms_page_store`

**Comment by lukeromanowicz** [comment]
I'll push a PR with a fix later tonight.

**Comment by kiber** [comment]
Thanks for the report, the bug was fixed in #76 and is already released as version 2.8.4
